WalletWasabi/BitcoinCore/Monitoring/P2pReconnector.cs

```diff
@@ -23,12 +23,12 @@ protected override async Task ActionAsync(CancellationToken cancel)
 			try
 			{
 				Logger.LogInfo("Trying to reconnect to P2P...");
-				P2pNode.Disconnect();
+				await P2pNode.DisconnectAsync(cancel).ConfigureAwait(false);
 				await P2pNode.ConnectAsync(cancel).ConfigureAwait(false);
 			}
 			catch
 			{
-				P2pNode.Disconnect();
+				await P2pNode.DisconnectAsync(cancel).ConfigureAwait(false);
```

Comment:
This `catch` will be called even when user uses cancellation token, right? Should `OperatationCanceledException` be handled separately?

Reply:
Hm, is that cancel scenario correct? Let's say that in `try` section you will attempt to disconnect, user cancels operation, you will get to `catch` clause but that cancel token is still canceled so you will not disconnect the node. Is this intended behavior?

Reply:
@kiminuo That was my issue, too: https://github.com/zkSNACKs/WalletWasabi/pull/4707/files/9f55fa1fedb9d10e10f4e0ce1412c75452877157#r520763464

And indeed [your solution](https://github.com/zkSNACKs/WalletWasabi/pull/4707#discussion_r522217959) while helps with this (at least don't call disconnect if disconnect fails/cancelled) but it does matter if the disconnection is cancelled or the connection.  

I'm not sure how to solve it though. Any idea?
 
Also @molnard are you planning to add @kiminuo's suggestion? https://github.com/zkSNACKs/WalletWasabi/pull/4707#discussion_r522217959

Reply:
Yes, I added `P2pNode` has

`public async Task<bool> TryDisconnectAsync(CancellationToken cancel)`

Reply:
I was not planning to refactor this but we went into it so I simplified this a bit. Pls take a look. 
Basically `P2pNode.Stop` cancels everything so do the `P2pReconnector.ActionAsync` and `P2pReconnector.StartAndAwaitReconnectionAsync()`

So we don't need to worry about this at the Disposal of P2pNode, neither we did before.